test/e2e/customconfigs/waitforpodsready_test.go

```diff
@@ -0,0 +1,302 @@
+					BackoffLimitCount:  ptr.To(int32(1)),
+				},
+			}
+
+			util.ApplyKueueConfiguration(ctx, k8sClient, cfg)
+			util.RestartKueueController(ctx, k8sClient)
+		})
+
+		ginkgo.By("creating a job that will never report its pods as ready", func() {
+			job = testingjob.MakeJob("job-timeout", ns.Name).
+				Queue(lq.Name).
+				Request(corev1.ResourceCPU, "2").
+				Parallelism(1).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, job)).Should(gomega.Succeed())
+		})
+
+		wlKey = types.NamespacedName{
+			Name:      workloadjob.GetWorkloadNameForJob(job.Name, job.UID),
+			Namespace: ns.Name,
+		}
+
+		workload := &kueue.Workload{}
+		ginkgo.By("waiting for the workload to be created and verifying it is not admitted", func() {
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, wlKey, workload)).Should(gomega.Succeed())
+
+				g.Expect(workload.Status.Admission).To(gomega.BeNil())
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		ginkgo.By("waiting for the workload to be evicted", func() {
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, wlKey, workload)).Should(gomega.Succeed())
+
+				cond := apimeta.FindStatusCondition(workload.Status.Conditions, kueue.WorkloadPodsReady)
+				g.Expect(cond).ShouldNot(gomega.BeNil())
+				g.Expect(cond.Status).To(gomega.Equal(metav1.ConditionFalse))
+				g.Expect(cond.Reason).To(gomega.Equal(kueue.WorkloadWaitForStart))
+
+				evictedCond := apimeta.FindStatusCondition(workload.Status.Conditions, kueue.WorkloadEvicted)
+				g.Expect(evictedCond).ShouldNot(gomega.BeNil())
+				g.Expect(evictedCond.Status).To(gomega.Equal(metav1.ConditionTrue))
+				g.Expect(evictedCond.Reason).To(gomega.Equal(kueue.WorkloadEvictedByPodsReadyTimeout))
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		ginkgo.By("verifying that the job is suspended", func() {
+			updatedJob := &batchv1.Job{}
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, types.NamespacedName{Name: job.Name, Namespace: ns.Name}, updatedJob)).Should(gomega.Succeed())
+				g.Expect(ptr.Deref(updatedJob.Spec.Suspend, false)).To(gomega.BeTrue())
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		ginkgo.By("verifying that the workload is requeued", func() {
+			gomega.Eventually(func(g gomega.Gomega) {
+				var wl kueue.Workload
+				g.Expect(k8sClient.Get(ctx, wlKey, &wl)).Should(gomega.Succeed())
+				g.Expect(wl.Status.RequeueState).ShouldNot(gomega.BeNil())
+				g.Expect(*wl.Status.RequeueState.Count).To(gomega.Equal(int32(1)))
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		ginkgo.By("simulating a second eviction cycle to check workload deactivation", func() {
+			var wl kueue.Workload
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, wlKey, &wl)).Should(gomega.Succeed())
+				if wl.Status.RequeueState != nil && *wl.Status.RequeueState.Count > 1 {
```

Comment:
```suggestion
				if wl.Status.RequeueState != nil && *wl.Status.RequeueState.Count > 1 {
```

Use g.Expect instead.